src/services/codefixes/helpers.ts

```diff
@@ -777,10 +796,40 @@ function createMethodImplementingSignatures(
     );
 }
 
+function getParameters(parameters: NodeArray<ParameterDeclaration>, signature: Signature, checker: TypeChecker, enclosingDeclaration: Node | undefined, flags: NodeBuilderFlags, context: TypeConstructionContext) {
+    return parameters.map((parameter, i) =>
+        parameter.type
+            && (!signature.thisParameter && i === 0)
+            && typeToTypeNode(checker, checker.getParameterType(signature, i - (signature.thisParameter ? 1 : 0)), enclosingDeclaration, flags, context).isTruncated

```

Comment:
I don't think I understand why we need to round-trip from a TypeNode (`parameter.type`) to a Type back to a TypeNode 🤔

Reply:
The only method I have found to make the checker re-serialize is by using `checker.typeToTypeNode`. Since `parameter.type` is `TypeNode`, I first get the type, then the TypeNode after identifying truncation.

Hope this makes sense.

Reply:
But why do we need to reserialize?

Reply:
I think the problem is that we're calling `checker.signatureToSignatureDeclaration` on this line: https://github.com/microsoft/TypeScript/pull/59332/files#diff-880eafae481c815c3af893123092cc0d93305284d8e6d2200638e7d546123b6cR414.

That means that the types get converted to type nodes as part of that call to `signatureToSignatureDeclaration`, and if we don't have direct control here of where the type nodes are being created, we can't replace the truncated ones.

I think a better approach to avoid serializing parameter types twice (and any other types that show up in a signature) would be to directly change the truncation behavior of the checker functions that convert things to nodes, e.g. `typeToTypeNode`, etc. So instead of inserting `...` and similar, it could insert `any`, for instance.
I think we could have a new context flag to control this, and set this new flag everywhere the relevant checker functions are being called in this file and others modified by this PR.

Reply:
You're correct @gabritto. The first time serializing is during `checker.signatureToSignatureDeclaration`. Replacing truncation there is going to be complicated becuase reporting truncation happens on high level. 

I do agree that is better to replace the `NoTruncation` behavior with using an `any` node. The challenge I see there is that by the time we have identified truncation, we have already serialized a bunch of nodes. We must then replace all of the with a single `any` node. Otherwise, we'll end up with a huge union with with an `any` in the second to last node.

I have not analyzed if that's feasible, but I can certainly look into it if it's promising.

Reply:
I was also wondering if the node builder could be configured to set a flag/property on node hierarchies that contain truncation.

Reply:
As per @gabritto comment, I've looked into adding a new flag that replaces the truncation behavior for replacing it with an `any` keyword. So far, the change looks good and very promising. The only problem is that we no longer have space for the NodeBuilderFlag to add a flag; I'm starting to look into that.

Reply:
I'm slightly concerned around `any` showing up in some of these places in case a user ever asks "where did this `any` come from?"

One alternative might be to create a node like `{ /*...*/ }` which retains that comment, or even `[...elided]`.

Another approach might be to introduce a special `...` type node that is recognized by the parser. It could still be an error, but gracefully parsed and interpreted as an `errorType` in the checker.

Maybe the concern with the `any` isn't as big a deal as I'm making it out to be, but I'm wondering what others think.

Reply:
I don’t think we should insert a syntax error. I think adding an inline comment like `/* elided */` or `/* error */` to an `any` is fine, maybe an improvement over a silently introduced `any`. I think a new special error type node is an interesting idea and I’m not against it; however, remember that this bug comes from fuzzing, not user reports or even telemetry, so I’m not sure how much effort should go into polishing what seems to be a rare edge case.

Reply:
I kind of like the idea of prepending a comment like `/* elided */` or something similar just to indicate to the user that this node got truncated and has been transformed to `any`. Arguably this might be confusing to the user, but at least something to search.

I have not search in telemetry to figure out the recurrence of this bug in release and I don't think we will get many user reports either because this causes completions to not be displayed. Maybe users didn't notice or care to report. Nonetheless, the error-deltas recurrently finds these errors.

Reply:
I have updated the PR to instead elide in case of truncation. This doesn't add a new flag as we discussed, because I think it would be unnecessary. Plus. we already handle it like that when creating declarations, so this is consistent with that behavior.

Ping @weswigham as there's some discussion about it privately.